### PR TITLE
Migrate to user tier

### PR DESCRIPTION
**Pull request by rajivnathan**
Related Issue: https://issues.redhat.com/browse/CRT-1363

Related PRs:
- https://github.com/codeready-toolchain/toolchain-common/pull/258
- https://github.com/codeready-toolchain/toolchain-e2e/pull/559

Summary of changes:
- Migrates existing MasterUserRecords to use UserTiers
- New MasterUserRecords to set the default UserTier instead of NSTemplateTier
- Removed ChangeTierRequest controller
- Updated other tests to use UserTiers instead of NSTemplateTiers

### CODE REVIEW

File: controllers/usersignup/masteruserrecord_test.go [hunk 1]
```diff
@@ -32,75 +41,109 @@ func TestMigrateMurIfNecessary(t *testing.T) {
...
-			assert.Equal(t, newExpectedMur(userSignup), mur)
-		})
-
 		t.Run("when tierName is missing", func(t *testing.T) {
 			userSignup := commonsignup.NewUserSignup()
-			nsTemplateTier := tiertest.NewNSTemplateTier("advanced", "dev", "stage", "extra")
-			mur := newMasterUserRecord(userSignup, test.MemberClusterName, nsTemplateTier, "johny")
+			defaultUserTier := testusertier.NewUserTier("deactivate90", 90)
+			mur := newMasterUserRecord(userSignup, test.MemberClusterName, "deactivate90", "johny")
 			mur.Spec.TierName = "" // tierName not set
```
Comment (MatousJobanek) [inline controllers/usersignup/masteruserrecord_test.go:67]: why not just
```suggestion
			mur := newMasterUserRecord(userSignup, test.MemberClusterName, "", "johny") // tierName not set
```

File: go.mod [hunk 2]
```diff
@@ -27,4 +27,6 @@ require (
 	sigs.k8s.io/controller-runtime v0.10.3
 )
 
+replace github.com/codeready-toolchain/toolchain-common => github.com/rajivnathan/toolchain-common v0.0.0-20220509204359-202bd6e094af
```
Comment (MatousJobanek) [inline go.mod:30]: reminder

**Review by MatousJobanek — APPROVED** [review]
Good job :+1: 
When are you going to remove the NSTemplateTier support from the deactivation controller? in the following PR?

**Comment by rajivnathan** [comment]
> Good job 👍 When are you going to remove the NSTemplateTier support from the deactivation controller? in the following PR?

Yes, exactly. Thanks for the review!

**Comment by rajivnathan** [comment]
/retest
should be fixed by previous e2e test fix

**Pull request merged**